server/service/service.go

```diff
@@ -145,11 +146,11 @@ func (s *Service) GetBootstrapData(ctx context.Context, req *bpb.GetBootstrapDat
 	log.Infof("Response set")
 
 	// Sign the response if Nonce is provided.
-	if req.Nonce != "" {
+	if req.GetNonce() != "" {
 		log.Infof("=============================================================================")
 		log.Infof("====================== Signing the response with nonce ======================")
 		log.Infof("=============================================================================")
-		resp.SignedResponse.Nonce = req.Nonce
+		resp.GetSignedResponse().Nonce = req.Nonce
```

Comment:
req.GetNonce() here as well

Reply:
In this case, GetNonce() returns the string by value, not by reference so it cannot be written to.

Reply:
resp.GetSignedResponse().Nonce = req.GetNonce()

Reply:
Ah, sorry missed that.